ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1864,19 +1885,29 @@ - (instancetype)copyWithZone:(NSZone *)zone {
 }
 
 - (BOOL)isAnswerValid:(id)answer {
-    BOOL isValid = NO;
     if ([answer isKindOfClass:[NSString class]]) {
         return [self isAnswerValidWithString:(NSString *)answer];
     }
-    return isValid;
+    return NO;
 }
 
 - (BOOL)isAnswerValidWithString:(NSString *)text {
-    if (text.length > 0) {
-        return ([self isTextLengthValidWithString:text] && [self isEmailAddressValidWithString:text]);
+    if (text) {
+        BOOL isValid = NO;
+        if (self.regex) {
+            NSString *regExPattern = self.regex;
+            NSRegularExpression *regEx = [[NSRegularExpression alloc] initWithPattern:regExPattern options:NSRegularExpressionCaseInsensitive error:nil];
+            NSUInteger regExMatches = [regEx numberOfMatchesInString:text options:0 range:NSMakeRange(0, [text length])];
+            isValid = (regExMatches != 0);
+        } else if (text.length > 0) {
+	        isValid = ([self isTextLengthValidWithString:text] && [self isEmailAddressValidWithString:text]);
```

Comment:
We can remove `[self isEmailAddressValidWithString:text]` from here and as well as it's implementation, since we are going to use the Validation Text initializer to validated email.